### ISSUE TITLE
question about 06_using_a_multiple_layer_network.py

### ISSUE THREAD

**Issue by WayneOuyang** [nfmcclure/tensorflow_cookbook#37]
Hello! I have a question about the mlp network which has around 200 parameters. But I check the sample size is less than 200. Depending on the seeds of random number, it's able to get converged results sometimes. my question is that how is it possible and is it the optimal model to run regression analysis to get the birth weight? Thanks!

**Comment by nfmcclure** [comment]
Hi @WayneOuyang ,  Thanks for the question.  Neural networks are not guaranteed to converge to the same solution everytime, as compared to a simple linear regression.  Because of this you can imagine multiple solutions (sets of parameters) that result in the same predictions.

The example dataset was for illustration, and of course is not ideal.  If I were to choose a model for this dataset, logistic regression would be better.  And if you choose a non-linear model (like we have), introducing regularization is a great idea (dropout).

Hope that helps!

-Nick

**Comment by WayneOuyang** [comment]
Hi @nfmcclure  One more question. Why do you normalize the input data to [0,1] range instead of zero mean and unit variance? Thanks!

**Comment by leigaoyi** [comment]
I run this example successfully too, but not very beautiful . Before I read nfmcclure's answer, I want to open an issue to question about the large loss in 06_using_a_multiple_.. , now I decide to optimize the solution by myself. As I understand the details about how to build a neural network from the example now. Thanks for this tutorial.

**Comment by nfmcclure** [comment]
Hi @WayneOuyang , Both normalizations should be sufficient for optimization.  Commonly you are supposed to transform the data to be centered around the highest non-linearity region of the activation function.  Both transformations should be sufficient for convergence.  I think that it would be user preference.
